packages/core/src/transport.ts

```diff
@@ -131,9 +131,10 @@ export class Batch<T> {
     return this.bufferMessageCount === this.maxSize || this.bufferBytesSize >= this.bytesLimit
   }
 
-  private sizeInBytes(candidate: string) {
-    // tslint:disable-next-line no-bitwise
-    return ~-encodeURI(candidate).split(/%..|./).length
+  private maxSizeInBytes(candidate: string) {
+    // the candidate byte size, once URI encoded can be maximized by 4 time its length
+    // Using this maximized value prevent any costly calculation while validating the byte size requirement
+    return candidate.length * 4
```

Comment:
FMU, in general case bytes size = length, so with this change we are going to generate 4 times more requests to the intake, I am not sure they will be ok with that.
could a more optimist estimation work? like +10% of the size

Reply:
We would generate 4 times more requests if all the batch are 4000+ characters. For a 3000 long batch, it is still under the 16kb limit.